src_python/habitat_sim/simulator.py

```diff
@@ -26,19 +26,260 @@
 except ImportError:
     _HAS_TORCH = False
 
-import habitat_sim.errors
+from habitat_sim import errors, gfx, scene, simulator
 from habitat_sim.agent.agent import Agent, AgentConfiguration, AgentState
 from habitat_sim.bindings import cuda_enabled
 from habitat_sim.logging import LoggingContext, logger
 from habitat_sim.metadata import MetadataMediator
 from habitat_sim.nav import GreedyGeodesicFollower, NavMeshSettings
-from habitat_sim.sensor import SensorSpec, SensorType
-from habitat_sim.sensors.noise_models import make_sensor_noise_model
+from habitat_sim.sensor import (  # Sensor, # TODO refactoring Sensor class
+    SensorSpec,
+    SensorType,
+)
+from habitat_sim.sensors.noise_models import SensorNoiseModel, make_sensor_noise_model
 from habitat_sim.sim import SimulatorBackend, SimulatorConfiguration
 from habitat_sim.utils.common import quat_from_angle_axis
 
 # TODO maybe clean up types with TypeVars
-ObservationDict = Dict[str, Union[bool, np.ndarray, "Tensor"]]
+ObservationTypeVar = Union[bool, ndarray, "Tensor"]
+ObservationDict = Dict[str, ObservationTypeVar]
+
+
+class SensorWrapper:
```

Comment:
Would inheriting from the SensorWrapper make more sense?

Reply:
Yes possibly. Currently I am just using the existing structure from the previous PR that I am continuing from and making sure it works after a merge. Then I will improve the code based on the direction we decide to go with regard to the sensor refactoring